examples/bump-alloc-tests/tests/tests.rs

```diff
@@ -0,0 +1,139 @@
+//! Tests that execute the entrypoints in the `bump-alloc-tests` contract, which
+//! doesn't have any functionality besides entrypoint that exercise and test the
+//! bump allocator from `concordium_std`.
+use concordium_smart_contract_testing::*;
+use concordium_std::collections::BTreeMap;
+
+const ACC_0: AccountAddress = AccountAddress([0u8; 32]);
+const ACC_INITIAL_BALANCE: Amount = Amount::from_ccd(1000);
+const SIGNER: Signer = Signer::with_one_key();
+
+/// Tests that the allocator correctly handles a data structure such as a map
+/// and that lookups return the correct values. Takes as input a map<u64, u64>,
+/// two keys, and the expected sum of corresponding values in the map.
+#[test]
+fn test_memory_layout() {
+    let (chain, contract_address) = initialize_chain_and_contract();
+
+    let map: BTreeMap<u64, u64> =
+        BTreeMap::from([(1, 2), (2, 3), (4, 5), (99, 100), (9999, 10000)]);
+    let key_1: u64 = 4;
+    let key_2: u64 = 9999;
+    let expected_sum: u64 = 5 + 10000;
+
+    chain
+        .contract_invoke(
+            ACC_0,
+            Address::Account(ACC_0),
+            Energy::from(10000),
+            UpdateContractPayload {
+                amount:       Amount::zero(),
+                address:      contract_address,
+                receive_name: OwnedReceiveName::new_unchecked(
+                    "bump_alloc_tests.memory_layout".to_string(),
+                ),
+                message:      OwnedParameter::from_serial(&(map, key_1, key_2, expected_sum))
+                    .expect("Parameter size is below limit."),
+            },
+        )
+        .print_emitted_events()
+        .expect("Update should succeed");
+}
+
+/// Test that allocating a large memory chunk the size of the maximum parameter
+/// length works.
+#[test]
+fn test_max_parameter_len() {
+    let (chain, contract_address) = initialize_chain_and_contract();
+
+    chain
+        .contract_invoke(
+            ACC_0,
+            Address::Account(ACC_0),
+            Energy::from(100_000),
+            UpdateContractPayload {
+                amount:       Amount::zero(),
+                address:      contract_address,
+                receive_name: OwnedReceiveName::new_unchecked(
+                    "bump_alloc_tests.max_parameter_len".to_string(),
+                ),
+                message:      OwnedParameter::from_serial(&[42u8; 65535])
+                    .expect("Parameter size is below limit."),
+            },
+        )
+        .print_emitted_events()
+        .expect("Update should succeed");
+}
```

Comment:
Could you please add a test where you 
- have some fixed static string
- allocate some non-trivial amount, say 500kB of data
- make a function call with non-trivial parameters, and that function does something with the parameters
- recurse
- do something more in the function with the static string and parameters or some locals in the function defined before the recursive call.

I want us to have a test to make sure that we are not overwriting the stack or the static data in the allocator.

Reply:
Yes, I've added that now. Let me know if you deem it sufficient ;)